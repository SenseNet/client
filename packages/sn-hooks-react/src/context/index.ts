export * from './current-ancestors'
export * from './current-children'
export * from './current-content'
export * from './injector'
export * from './load-settings'
export * from './logger'
export * from './repository'
export * from './session'
