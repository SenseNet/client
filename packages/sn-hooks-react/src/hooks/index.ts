export * from './use-authentication'
export * from './use-download'
export * from './use-injector'
export * from './use-logger'
export * from './use-repository-events'
export * from './use-repository'
export * from './use-session'
export * from './use-version-info'
export * from './use-wopi'
