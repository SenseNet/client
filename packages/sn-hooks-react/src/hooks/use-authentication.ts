import { authenticationResultStatus, AuthFuncResult, OIDCAuthenticationService } from '@sensenet/client-core'
import { useState } from 'react'
import { useLogger } from '.'

export interface AuthActionParams {
  returnUrl: string
  authService: OIDCAuthenticationService
  successCallback?: (url: string) => void
}

/**
 * useAuthentication hook helps provide helper methods for login and logout with an OIDC authentication service.
 * It **doesn't** work with forms authentication.
 */
export function useAuthentication() {
  const [isLoading, setIsLoading] = useState(false)
  const [error, setError] = useState<string>()
  const logger = useLogger('useAuthentication')

  const doAuthenticationAction = async (action: () => AuthFuncResult, successCallback?: (url: string) => void) => {
    setIsLoading(true)
    const result = await action()
    setIsLoading(false)
    switch (result.status) {
      case authenticationResultStatus.redirect:
        break
      case authenticationResultStatus.success:
        successCallback?.(result.state.returnUrl)
        break
      case authenticationResultStatus.fail:
        logger.warning({ message: result.error.message })
        setError(result.error.message)
        break
      default:
        logger.debug({ message: `Invalid status result ${result}.` })
    }
  }

  const login = async ({ authService, returnUrl, successCallback }: AuthActionParams) => {
    await doAuthenticationAction(() => authService.signIn({ returnUrl }), successCallback)
  }

  const logout = async ({ authService, returnUrl, successCallback }: AuthActionParams) => {
    await doAuthenticationAction(() => authService.signOut({ returnUrl }), successCallback)
  }

  return { login, isLoading, error, logout }
}
