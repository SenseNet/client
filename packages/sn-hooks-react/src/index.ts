export * from './context'
export * from './hooks'
export * from './oidc-routes'
