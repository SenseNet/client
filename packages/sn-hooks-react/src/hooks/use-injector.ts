import { useContext } from 'react'
import { InjectorContext } from '../context'

/**
 * Custom hook that returns an Injecor instance from the InjectorContext.
 */
export const useInjector = () => useContext(InjectorContext)
