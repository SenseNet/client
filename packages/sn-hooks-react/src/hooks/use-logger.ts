import { useContext, useState } from 'react'
import { LoggerContext } from '../context/LoggerContext'

/**
 * Returns a scoped Logger instance from the LoggerContext.
 * @param scope The logger's Scope name
 */
export const useLogger = (scope: string) => {
  const [logger] = useState(useContext(LoggerContext).withScope(scope))
  return logger
}
