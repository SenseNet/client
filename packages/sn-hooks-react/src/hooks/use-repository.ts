import { useContext } from 'react'
import { RepositoryContext } from '../context'

/**
 * Returns a Repository instance from the RepositoryContext
 */
export const useRepository = () => useContext(RepositoryContext)
