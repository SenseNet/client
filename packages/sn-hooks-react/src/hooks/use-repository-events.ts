import { useEffect, useState } from 'react'
import { EventHub } from '@sensenet/repository-events'
import { useRepository } from './use-repository'

export const useRepositoryEvents = () => {
  const repo = useRepository()
  const [eventHub, setEventHub] = useState(new EventHub(repo))

  useEffect(() => {
    const eh = new EventHub(repo)
    setEventHub(eh)
    return () => eh.dispose()
  }, [repo])

  return eventHub
}
