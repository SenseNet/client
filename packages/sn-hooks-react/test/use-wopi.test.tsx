import React from 'react'
import { shallow } from 'enzyme'
import { ActionModel, GenericContent } from '@sensenet/default-content-types'
import { useWopi } from '../src/hooks'

const WopiDump: React.FC<{ content: GenericContent }> = ({ content }) => {
  const i = useWopi(content)
  return <div>{JSON.stringify(i)}</div>
}

describe('Wopi', () => {
  it('Should match the snapshot', () => {
    expect(shallow(<WopiDump content={{ Id: 1, Path: '', Type: 'File', Name: 'aaa.txt' }} />)).toMatchSnapshot()
  })

  it('Should allow edit based on available action', () => {
    expect(
      shallow(
        <WopiDump
          content={{
            Id: 1,
            Path: '',
            Type: 'File',
            Name: 'aaa.txt',
            Actions: [{ Name: 'WopiOpenEdit' } as ActionModel],
          }}
        />,
      ),
    ).toMatchSnapshot()
  })

  it('Should allow view based on available action', () => {
    expect(
      shallow(
        <WopiDump
          content={{
            Id: 1,
            Path: '',
            Type: 'File',
            Name: 'aaa.txt',
            Actions: [{ Name: 'WopiOpenView' } as ActionModel],
          }}
        />,
      ),
    ).toMatchSnapshot()
  })
})
