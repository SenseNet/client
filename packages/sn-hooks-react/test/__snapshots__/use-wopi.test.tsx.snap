// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wopi Should allow edit based on available action 1`] = `
<div>
  {"isReadAwailable":{"Name":"WopiOpenEdit"},"isWriteAwailable":{"Name":"WopiOpenEdit"}}
</div>
`;

exports[`Wopi Should allow view based on available action 1`] = `
<div>
  {"isReadAwailable":{"Name":"WopiOpenView"}}
</div>
`;

exports[`Wopi Should match the snapshot 1`] = `
<div>
  {}
</div>
`;
